### PR TITLE
Read Excel Source

### PR DESCRIPTION
**Pull request by hgrif**

### CODE REVIEW

File: dbt/adapters/duckdb/relation.py [hunk 1]
```diff
@@ -34,6 +37,21 @@ def create_from_source(cls: Type[Self], source: ParsedSourceDefinition, **kwargs
             ext_location = ext_location.format(
                 schema=source.schema, name=source.name, identifier=source.identifier
             )
+            if ".xlsx" in ext_location:
+
+                # Strip quotes.
+                if ext_location.startswith("'"):
```
Comment (JCZuurmond) [inline dbt/adapters/duckdb/relation.py]: Why not use the 'strip' function?

File: dbt/adapters/duckdb/relation.py [hunk 2]
```diff
@@ -34,6 +37,21 @@ def create_from_source(cls: Type[Self], source: ParsedSourceDefinition, **kwargs
...
+                    ext_location = ext_location[1:-1]
+
+                df = pd.read_excel(ext_location)
+                TEMP_DIR = pathlib.Path("/tmp/dbt-excel/")
```
Comment (JCZuurmond) [inline dbt/adapters/duckdb/relation.py]: I was thinking that we could create a directory at the same location and with the same name as the Excel file

**Review by JCZuurmond — APPROVED** [review]
Nice we got this to work this quick! Let's review after the holidays if we are happy with this approach. I expect this to be the best approach (for now maybe you could write an Excel extension for DuckDB 😉)

**Comment by hgrif** [comment]
@JCZuurmond: I've implemented your suggestions

**Pull request merged**